docs/website/contents/for-developers/HardWonWisdom.md

```diff
@@ -348,3 +348,77 @@ The key corresponding parts of the implementation are as follows.
+                - Notably, this constraint must be true if the function is actually counting blocks in order to achieve its montonicity with respect to maximum rollbacks without relying on a Chain Growth property.
+
+See also "Which slots/times/epochs can we translate?" above.
+
+Some notable components are not listed above because they are determined by existing data about `F` and `G`.
+
+- The `TopLevelConfig (HardForkBlock [F, G])` can be completely determined from the same for `F` and `G`.
+- The initial state of the ledger and protocol can be completely determined from the same for `F`.
+
+### Eras that contain no slots
+
+The HFC interface and contract together permit `singleEraTransition cfg eraParams start st` to return `Just (boundEpoch start)`.
+In that case, the era contains no slots on the chain (and hence no blocks).
+It can only happen if `st` is the either the initial ledger state or else the result of translating the `LedgerState` from the latest block.
+The latest block isn't even necessarily in the preceding era, since that translation could have invoked multiple individual translation functions at once, if multiple eras contains no slots.
+
+It is not immediately obvious how this relates to the safe zone.
+Consider an era sequence `[F, G, H]`.
+If `G` has no slots, then the latest its end could have became known is once `F`'s end became known.
+Thus, the safe zone would seem to be respected as long as `G`'s safe zone is no larger than `F`'s.
+However, it may be too subtle that the block `x` in the definition of the safe zone is not necessarily from the current era!
+
+It's unclear if it's even desirable for the HFC to allow eras that contain no slots, with one exception: the Cardano system-level benchmarks already rely on this behavior in order to skip a prefix of the chain's eras.
+Indeed, skipping a prefix of eras seems like the most natural use case, since the "genesis block" itself is already not an actual block.
+But there is likely a more direct way to support that use case.
+And so it seems possible and desirable to simplify the HFC interface so that it only applies `singleEraTransition` to a ledger state arising from a block in the current era.
+Something alone the lines of "if an era's end is known before its first block exists, then the era must end before slot 0".
+
+TODO the current code (and Edsko's chapter in the report) sometimes applies the safe zone to the first slot of the epoch.
+How does that relate here?
+Perhaps it suggests that the translated-but-not-actually-ticked state is in fact considered to have been "ticked" up to the beginning of the next era.
+And so `singleEraTransition` is interpreting translated state from the "previous era" as a kind of summary of everything that came before, a la the "genesis block"?
```

Comment:
Maybe a big improvement would simply be renaming `translateLedgerState` and `Translate` to `convertToInitialLedgerState`/`reinterpretAsGenesisBlock`/etc.

Would that have made it more obvious that the current era's tick rule won't be used for the end of the epoch?

Reply:
I guess it would definitely have helped to signal (at least to me) that something weird is going on (in contrast to "translate", which sounds more principled/generic); but I don't know if these specific names would have helped, eg these functions don't reset the ledger state tip to `Origin` as one might expect in a vacuum.

Reply:
I'm leaving this TODO in--its appropriate content for `HardWonWisdom.md`